errdefs/internal.go

```diff
@@ -0,0 +1,54 @@
+package errdefs
+
+import (
+	"errors"
+	"slices"
+	"syscall"
+)
+
+type internalErr struct {
+	error
+}
+
+func (internalErr) System() {}
+
+func (err internalErr) Unwrap() error {
+	return err.error
+}
+
+type system interface {
+	System()
+}
+
+var _ system = internalErr{}
+
+var systemErrors = []syscall.Errno{
+	syscall.EIO,             // I/O error
+	syscall.ENOMEM,          // Out of memory
```

Comment:
@silvin-lubecki I'm not sure if we should mark `ENOMEM` and `ENOSPC` as internal like others or maybe some `ResourceExhausted` code would make more sense. Technically these are likely caused by too much usage from user, not internal bugs.

Reply:
I agree with you @tonistiigi. I like what you suggested with the specific error code, it allows some latitude to interpret it as user or internal error 👍

Reply:
Moved `ENOMEM` and `ENOSPC` to `ResourceExhausted` code instead.